apm-agent-core/src/test/java/co/elastic/apm/agent/impl/ElasticApmTracerTest.java

```diff
@@ -159,11 +160,20 @@ void testEnableStacktraces() throws InterruptedException {
             Span span = tracerImpl.getActive().createSpan();
             try (Scope spanScope = span.activateInScope()) {
                 Thread.sleep(10);
-                span.end();
+                stackTraceEndSpan(span);
             }
             transaction.end();
         }
-        assertThat(reporter.getFirstSpan().getStacktrace()).isNotNull();
+        Throwable stackTrace = reporter.getFirstSpan().getStacktrace();
+        assertThat(stackTrace).isNotNull();
+        assertThat(Arrays.stream(stackTrace.getStackTrace()).filter(stackTraceElement ->
+            stackTraceElement.getMethodName().equals("stackTraceEndSpan")
+                && stackTraceElement.getClassName().equals(ElasticApmTracerTest.class.getName()))).hasSize(1);
+    }
```

Comment:
[for reviewer] this test does not actually check that the bugfix is effective, but just improves a bit the heuristic because until now there was none.